src/src/Controller/CustomerSyncController.php

```diff
@@ -0,0 +1,266 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Smaily for PrestaShop. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author    Smaily <info@smaily.com>
+ * @copyright 2024 Smaily
+ * @license   GPL3
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\Module\SmailyForPrestaShop\Controller;
+
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+use PrestaShop\Module\SmailyForPrestaShop\Lib\Api;
+use PrestaShop\Module\SmailyForPrestaShop\Lib\Logger;
+use PrestaShop\Module\SmailyForPrestaShop\Model\Subscriber;
+use PrestaShop\Module\SmailyForPrestaShop\Model\SubscriberCollection;
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+
+class CustomerSyncController
+{
+    /**
+     * Limit unsubscribers request batch size.
+     */
+    private const UNSUBSCRIBERS_BATCH_LIMIT = 1000;
+
+    /**
+     * Limit subscribers query batch size.
+     */
+    private const SUBSCRIBERS_BATCH_LIMIT = 1000;
+
+    /**
+     * @var ConfigurationInterface
+     */
+    private $configuration;
+
+    /**
+     * @var Api;
+     */
+    private $api;
+
+    public function __construct(ConfigurationInterface $configuration)
+    {
+        $this->configuration = $configuration;
+
+        $subdomain = $configuration->get('SMAILY_SUBDOMAIN');
+        $username = $configuration->get('SMAILY_USERNAME');
+        $password = $configuration->get('SMAILY_PASSWORD');
+
+        if (!empty($subdomain) && !empty($username) && !empty($password)) {
+            $this->api = new Api($subdomain, $username, $password);
+        }
+    }
+
+    public function sync(string $token): bool
+    {
+        if ($token !== $this->configuration->get('SMAILY_CUSTOMER_CRON_TOKEN')) {
+            echo 'Access denied!';
+
+            return false;
+        }
+
+        if (!$this->configuration->get('SMAILY_ENABLE_CUSTOMER_SYNC')) {
+            echo 'User synchronization disabled!';
+
+            return false;
+        }
+
+        if (empty($this->api)) {
+            echo 'Account setup not finished!';
+
+            return false;
+        }
+
+        $unsubscribers_synchronized = $this->removeUnsubscribers(self::UNSUBSCRIBERS_BATCH_LIMIT);
+        if (!$unsubscribers_synchronized) {
+            echo 'Failed to synchronize unsubscribers from Smaily!';
+
+            Logger::logMessageWithSeverity('Customer sync failed - unsubscribers are not removed', 1);
+
+            return false;
+        }
+
+        // Don't sync customers if failed to remove unsubscribers.
+        $subscribers_synchronized = $this->sendSubscribersToSmaily(self::SUBSCRIBERS_BATCH_LIMIT);
+        if (!$subscribers_synchronized) {
+            echo 'Failed to synchronize subscribers from the store!';
+
+            Logger::logMessageWithSeverity('Customer sync failed - failed to send subscribers to Smaily', 1);
+
+            return false;
+        }
```

Comment:
This approach has one inherent flaw - Smaily will always be the source of truth for opt-outs when only using Customer Synchronization in the module. When synchronizing customers opt-outs from Smaily need to be compared against opt-in times in PrestaShop.

If this is not in the scope of the release, then we need to consider it in the future.

Reply:
The scope of this PR is already quite big.  I would like to push feature improvements to the following ones.

Reply:
Okay, let's push this to a future release.